Examples/Io/Root/src/RootTrackStatesWriter.cpp

```diff
@@ -289,8 +293,25 @@ ActsExamples::ProcessCode ActsExamples::RootTrackStatesWriter::finalize() {
   return ProcessCode::SUCCESS;
 }
 
-ActsExamples::ProcessCode ActsExamples::RootTrackStatesWriter::writeT(
-    const AlgorithmContext& ctx, const ConstTrackContainer& tracks) {
+RootTrackStatesWriter::StateType RootTrackStatesWriter::getStateType(
+    ConstTrackStateProxy state) {
+  if (state.typeFlags().test(Acts::MeasurementFlag)) {
+    return StateType::eMeasurement;
+  }
+  if (state.typeFlags().test(Acts::OutlierFlag)) {
+    return StateType::eOutlier;
+  }
+  if (state.typeFlags().test(Acts::HoleFlag)) {
+    return StateType::eHole;
+  }
+  if (state.typeFlags().test(Acts::MaterialFlag)) {
+    return StateType::eMaterial;
+  }
+  return StateType::eUnknown;
```

Comment:
More than one of the flags can be set (that's why it's not an enum).
In this case, I believe e.g. outliers will be written out as measurements.

So, maybe we should instead store each of them as individual booleans, so you can filter on them in the ntuple?

Reply:
I wondered if it is enough for this simple case to have an enum because they should be fairly exclusive here no?
- Outlier will always have a measurement but we rather care about it being an outlier
- Measurement when we accepted it
- Hole cannot have a measurement
- Material when it is a pure material state

Good point with the if checking measurement first. I can rotate that